src/controllers/baseHandler.ts

```diff
@@ -42,9 +46,27 @@ export async function baseHandler(
         roleToBeTaggedObj: data[0],
         displayMessageObj: data[1] ?? {},
       };
-
       return await mentionEachUser(transformedArgument, env);
     }
+
+    case getCommandName(LISTENING): {
+      const data = message.data?.options;
+      const setter = data ? data[0].value : false;
+      const nickname = removeListening(message.member.nick || "");
+      if (setter) {
+        if (!message.member.nick?.includes(LISTENING_COPY)) {
+          await updateNickName(
```

Comment:
plesae use async methods that make external calls in try catch also return an appropriate response to user reflecting an internal server error.